### PR TITLE
Add element to load a playlist from LB

### PR DESCRIPTION
**Pull request by amCap1712**
Add a method to de-serialize JSPF playlist to a `troi.Playlist` entity. Also, added a dummy patch for now but we may not want to put that in LB instead of troi or maybe go entirely some other way.

### CODE REVIEW

File: troi/patches/playlist_from_listenbrainz.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from troi.playlist import PlaylistFromJSPFElement
+import troi.musicbrainz.recording_lookup
+
+
+@click.group()
+def cli():
+    pass
+
+
+class ResavePatch(troi.patch.Patch):
```
Comment (amCap1712) [inline troi/patches/playlist_from_listenbrainz.py:13]: We may want to put this patch in LB or go some other way. Probably best to wait until troi is refactored to be a better library.

File: troi/playlist.py [hunk 2]
```diff
@@ -255,6 +284,11 @@ def submit_to_spotify(self, user_id: str, token: str, is_public: bool = True, is
             result = sp.playlist_add_items(spotify_playlist["id"], spotify_track_ids)
             submitted.append((spotify_playlist["external_urls"]["spotify"], spotify_playlist["id"]))
 
+            if playlist.external_urls:
```
Comment (amCap1712) [inline troi/playlist.py:287]: Need to decide on exact way how this should be stored in Playlist entity. We also likely want to store the unmatched tracks or least their count.

File: troi/patches/playlist_from_listenbrainz.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    @staticmethod
+    def description():
+        return "Retrieve a playlist from the ListenBrainz"
+
+    def create(self, inputs):
+        token = inputs.get("read_only_token")
+        if not token:
+            token = inputs.get("token")
+        playlist = PlaylistFromJSPFElement(inputs["mbid"], token)
```
Comment (mayhem) [inline troi/patches/playlist_from_listenbrainz.py:47]: could collapse into one line.

File: troi/playlist.py [hunk 4]
```diff
@@ -396,3 +431,32 @@ def read(self, inputs):
...
+
+    def read(self, inputs):
+        headers = None
+        if self.token:
+            headers = {"Authorization": f"Token {self.token}"}
+        response = requests.get(LISTENBRAINZ_PLAYLIST_FETCH_URL + self.playlist_mbid, headers=headers)
+        response.raise_for_status()
+        data = response.json()
+
+        playlist = _deserialize_from_jspf(data)
```
Comment (mayhem) [inline troi/playlist.py:461]: here too collapse into one line.

**Review by mayhem — APPROVED** [review]
I think the naming could be improved -- lets discuss in IRC.

File: troi/playlist.py (hunk 4 above)
Comment (amCap1712) [reply troi/playlist.py]: Done

**Pull request merged**